CMakeLists.txt

```diff
@@ -218,6 +217,14 @@ if(CMAKE_SYSTEM_NAME STREQUAL "SunOS")
   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -m64")
 endif()
 
+if(GIT_COMMIT)
+    add_compile_definitions(GIT_COMMIT="${GIT_COMMIT}")
+endif()
+
+if(GIT_BRANCH)
+    add_compile_definitions(GIT_BRANCH="${GIT_BRANCH}")
+endif()
+
```

Comment:
Why not to try to get them as before if they are not provided?
The worse scenario is that they could be empty.

Reply:
Maybe it could be as an extra development option

Reply:
IMO it is even better, we will know that it was not generated from our official repo.